test fails when running tests of flow-client and vaadin-spring modules simultaneously

### Description of the bug

When running `flow-client tests` they pass, it also happens for `vaadin-spring`, but running both simultaneously there is a test regarding serialisation that fails. I can reproduce it in mac, gitpod and circleci, but seems working in gihub action containers.


```
[ERROR] Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.545 s <<< FAILURE! - in com.vaadin.flow.spring.SpringClassesSerializableTest
[ERROR] classesSerializable  Time elapsed: 0.494 s  <<< ERROR!
java.lang.NoClassDefFoundError: elemental/dom/Element
        at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
        at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
        at java.base/java.lang.Class.privateGetPublicMethods(Class.java:3191)
        at java.base/java.lang.Class.getMethods(Class.java:1904)
        at com.vaadin.flow.testutil.ClassFinder.isTestClass(ClassFinder.java:149)
        at com.vaadin.flow.testutil.ClassFinder.isTestClass(ClassFinder.java:144)
        at com.vaadin.flow.testutil.ClassesSerializableTest.classesSerializable(ClassesSerializableTest.java:306)


```

### Expected behavior

I can test all flow modules simultaneously

### Minimal reproducible example

checkout flow master branch and run
```
mvn clean install -DskipTests
mvn test -pl flow-client,vaadin-spring
```


### Versions

- Vaadin / Flow version: 23.1-SNAPSHOT
- Java version: 11
- OS version: macos BS

The problem seems to be that when running test phase on the two modules together, `flow-client/target/classes` gets added to  `java.class.path` also for `vaadin-spring` module, so some classes the test is trying to load misses dependencies.

I have locally an error similar to the one described above, because the test is loading `com.vaadin.flow.linker.ClientEngineLinker`

```
==================================== Loading class com.vaadin.flow.linker.ClientEngineLinker
[ERROR] Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.177 s <<< FAILURE! - in com.vaadin.flow.spring.SpringClassesSerializableTest
[ERROR] classesSerializable  Time elapsed: 0.605 s  <<< ERROR!
java.lang.NoClassDefFoundError: com/google/gwt/core/linker/SingleScriptLinker
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:315)
	at com.vaadin.flow.testutil.ClassesSerializableTest.classesSerializable(ClassesSerializableTest.java:314)
```


Perhaps we can just add `com.vaadin.flow.linker` and `com.vaadin.client` to the list of the exclusion patterns in `ClassesSerializableTest`, since classes in those packages should be used at compile time only, if I remember correctly.

--- a/flow-test-generic/src/main/java/com/vaadin/flow/testutil/ClassesSerializableTest.java
+++ b/flow-test-generic/src/main/java/com/vaadin/flow/testutil/ClassesSerializableTest.java
@@ -207,6 +207,11 @@ public abstract class ClassesSerializableTest extends ClassFinder {
                 "com\\.vaadin\\.flow\\.server\\.frontend\\.AbstractTaskClientGenerator",
                 "com\\.vaadin\\.flow\\.server\\.frontend\\.EndpointGeneratorTaskFactory",
 
+                // Flow client classes
+                "com\\.vaadin\\.client\\..*",
+                "com\\.vaadin\\.flow\\.linker\\.ClientEngineLinker",
+                "com\\.vaadin\\.flow\\.linker\\.ClientEngineLinker\\$Script",
+
                 // Node downloader classes
                 "com\\.vaadin\\.flow\\.server\\.frontend\\.installer\\.DefaultArchiveExtractor",
                 "com\\.vaadin\\.flow\\.server\\.frontend\\.installer\\.ArchiveExtractor",

